src/main/java/teammates/logic/core/GoogleCloudLoggingService.java

```diff
@@ -119,16 +121,29 @@
     }
 
     @Override
-    public QueryLogsResults queryLogs(List<String> severities, Instant startTime, Instant endTime,
-            Integer pageSize, String pageToken) throws LogServiceException {
+    public QueryLogsResults queryLogs(QueryLogsParams queryLogsParams)
+            throws LogServiceException {
+
         LogSearchParams logSearchParams = new LogSearchParams()
                 .addLogName(STDOUT_LOG_NAME)
                 .addLogName(STDERR_LOG_NAME)
-                .setSeverities(severities)
-                .setStartTime(startTime)
-                .setEndTime(endTime);
+                .setStartTime(queryLogsParams.getStartTime())
+                .setEndTime(queryLogsParams.getEndTime())
+                .setTraceId(queryLogsParams.getTraceId())
+                .setActionClass(queryLogsParams.getActionClass())
+                .setUserInfoParams(queryLogsParams.getUserInfoParams())
+                .setLogEvent(queryLogsParams.getLogEvent())
+                .setSourceLocation(queryLogsParams.getSourceLocation())
+                .setExceptionClass(queryLogsParams.getExceptionClass());
```

Comment:
Looking at the number of times you simply pass on the information without any modification, do you really need different `LogSearchParams` and `QueryLogsParams`?

Reply:
One reason that we use a different param class is that `LogSearchParams` is a private nested class. Unless we make it public, otherwise we need another way to pass around the parameters.

Reply:
> `LogSearchParams` is a private nested class

"Private", "Nested", etc. are implementation details that are simple to change. The question is not answered: do you really need different `LogSearchParams` and `QueryLogsParams`?